peg/elm_post.h

```diff
@@ -0,0 +1,155 @@
+/*
+ *   Copyright (c) 2022 Nik Silver
+ *   based on the Kotlin code by Jan Dolinár
+ *
+ *   This source code is released for free distribution under the terms of the
+ *   GNU General Public License version 2 or (at your option) any later version.
+ *
+ *   This module contains functions to generate tags for Elm.
+ */
+
+/*
+ *   INCLUDE FILES
+ */
+#include "parse.h"
+#include "trace.h"
+
+/*
+ * FUNCTION DEFINITIONS
+ */
+
+/* kind - The kind of this tag.
+ * role - The role this tag plays. But def(ined) is expressed by setting
+ *     this to ROLE_DEFINITION_INDEX.
+ * pushScope - If true, also update the scope to be this tag.
+ */
+static int makeElmTag (struct parserCtx *auxil, const char *name, long offset, int kind, int role)
+{
+	tagEntryInfo e;
+
+	if (role == ROLE_DEFINITION_INDEX)
+	{
+		initTagEntry (&e, name, kind);
+	}
+	else
+	{
+		initRefTagEntry (&e, name, kind, role);
+	}
+
+	e.lineNumber = getInputLineNumberForFileOffset (offset);
+	e.filePosition = getInputFilePositionForLine (e.lineNumber);
+	e.extensionFields.scopeIndex = BASE_SCOPE (auxil);
+	int scope_index = makeTagEntry (&e);
+	return scope_index;
+}
+
+/* Like makeElmTag, but have this tag be the latest scope.
+ */
+static int makeElmTagSettingScope (struct parserCtx *auxil, const char *name, long offset, int kind, int role)
+{
+	// Here is a useless line just to use a function and so
+	// get rid of a compiler warning.
+	if (1 < 0)
+	{
+		PEEK_KIND (auxil);
+	}
+	// The real function starts here...
+
+	int scope_index = makeElmTag (auxil, name, offset, kind, role);
+	SET_SCOPE (auxil, scope_index);
+	return scope_index;
+}
+
+static void addElmSignature(int scope_index, const char *sig)
+{
+	tagEntryInfo *e = getEntryInCorkQueue (scope_index);
+
+	if (e)
+	{
+		vString *vsig = collapseWhitespace (sig);
+		const char *sig2 = vStringValue (vsig);
+
+		e->extensionFields.signature = eStrdup (sig2);
+
+		vStringDelete (vsig);
```

Comment:
For this popular code pattern, there is a helper function.

```C
		vString *vsig = collapseWhitespace (sig);
		const char *sig2 = vStringValue (vsig);

		e->extensionFields.signature = eStrdup (sig2);

		vStringDelete (vsig);
```
You can write this like:
```
		vString *vsig = collapseWhitespace (sig);
		e->extensionFields.signature = vStringDeleteUnwrap(vsig);
```
You can reduce two "malloc":-)